pydrive2/files.py

```diff
@@ -97,6 +98,48 @@ def _GetList(self):
         return result
 
 
+class FakeWriteable(object):
+    def write(self, chunk):
+        self.chunk = chunk
+
+
+class MediaIoReadable(object):
+    def __init__(self, request, chunksize=DEFAULT_CHUNK_SIZE, pre_buffer=True):
+        """File-like wrapper around MediaIoBaseDownload.
+
+    :param pre_buffer: Whether to read one chunk into an internal buffer
+    immediately in order to raise any potential errors.
+    :raises: HttpError
+    """
+        self.done = False
+        self.fd = FakeWriteable()
+        self.downloader = MediaIoBaseDownload(
+            self.fd, request, chunksize=chunksize
+        )
+        self._buffer = None
+        if pre_buffer:
+            self._buffer = self.read()
+
+    def read(self, chunksize=DEFAULT_CHUNK_SIZE):
+        """
+    :returns: str -- chunk or None if done
+    :raises: ApiRequestError
+    """
+        if self._buffer:
+            buffer = self._buffer
+            self._buffer = None
+            return buffer
+        if self.done:
+            return None
+        if chunksize:
+            self.downloader._chunksize = chunksize
+        try:
+            _, self.done = self.downloader.next_chunk()
+        except errors.HttpError as error:
+            raise ApiRequestError(error)
+        return self.fd.chunk
```

Comment:
can it be empty in an edge case? is it valid?

Reply:
empty should be fine and valid. As I noted in the docstring, really it is up to the user to check against `None` whether it's done. Though logically I see no reason why an empty string would be returned, or for that matter a string less than `chunksize` if we're not yet done.